File: Modules/CustomerManagement/CustomerActivity/CustomerActivityColumns.cs
namespace MapPin.CustomerManagement.Forms;

[ColumnsScript("CustomerManagement.CustomerActivity")]
[BasedOnRow(typeof(CustomerActivityRow), CheckNames = true)]
public class CustomerActivityColumns
{
    [EditLink, SortOrder(1, descending: true)]
    public DateTime ActivityDate { get; set; }

    [EditLink, Width(120)]
    public CustomerActivityType ActivityType { get; set; }

    [EditLink]
    public string Subject { get; set; }

    public string Notes { get; set; }
}

File: Modules/ServerTypes/CustomerManagement/CustomerActivityColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CustomerActivityRow } from "./CustomerActivityRow";
import { CustomerActivityType } from "./CustomerActivityType";

export interface CustomerActivityColumns {
    ActivityDate: Column<CustomerActivityRow>;
    ActivityType: Column<CustomerActivityRow>;
    Subject: Column<CustomerActivityRow>;
    Notes: Column<CustomerActivityRow>;
}

export class CustomerActivityColumns extends ColumnsBase<CustomerActivityRow> {
    static readonly columnsKey = 'CustomerManagement.CustomerActivity';
    static readonly Fields = fieldsProxy<CustomerActivityColumns>();
}

[CustomerActivityType]; // referenced types